### ISSUE TITLE
ORKGraphChartView / xAxis / Color Problem

### ISSUE THREAD

**Issue by milifili** [ResearchKit/ResearchKit#622]
I just add the following line into ORKCatalog sample project to change the global appearance of UIView : 
```
func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
        
        UIView.appearance().tintColor = UIColor.greenColor(); // <==== this is the added line
        // When a task result has been finished, update the result view controller's task result.
        taskListViewController.taskResultFinishedCompletionHandler = { [unowned self] taskResult in
            /*
                If we're displaying a new result, make sure the result view controller's
                navigation controller is at the root.
            */
            if let navigationController = self.resultViewController?.navigationController {
                navigationController.popToRootViewControllerAnimated(false)
            }
            
            // Set the result so we can display it.
            self.resultViewController?.result = taskResult
        }
        
        return true
    }
```
The result: 
![simulator screen shot feb 23 2016 18 06 13](https://cloud.githubusercontent.com/assets/5347193/13261503/40b00c34-da58-11e5-8d51-3957d75ef436.png)

By the way, I think the solution is as easy as adding a line to ORKGraphChartView.m file : 
```
- (void)setDataSource:(id<ORKGraphChartViewDataSource>)dataSource {
    _dataSource = dataSource;
    _numberOfXAxisPoints = -1; // reset cached number of x axis points
    [self updateAndLayoutVerticalReferenceLineLayers];
    [self obtainDataPoints];
    [self calculateMinAndMaxValues];
    _xAxisView.tintColor = self.tintColor; // <==== this line should be added
    [_xAxisView updateTitles];
    [_yAxisView updateTicksAndLabels];
    [self updateLineLayers];
    [self updatePointLayers];
    [self updateNoDataLabel];
    
    [self setNeedsLayout];
}
```

**Comment by rsanchezsaez** [comment]
Is it your issue that you want the `ORKGraphChartView` not to use the global tint color for the horizontal axis titles? If that's the case, you can manually set the desired `tintColor` to your `ORKGraphChartView` instance (which can be different from the global tint color).

Additionally, the *vertical axis title color* can be customized through the `verticalAxisTitleColor` property.

**Comment by milifili** [comment]
Thanks for reply. If I set red color to property you said, the xAxis labels color doesn't change and remains as same as Global tint color!

    self.graphView.axisColor = [UIColor redColor];
    self.graphView.tintColor =   [UIColor redColor];
    self.graphView.verticalAxisTitleColor = [UIColor redColor];`

Only the xAxis line changes not labels! Please try to change xAxis label.

**Comment by rsanchezsaez** [comment]
Oh, you are right! This is a bug. Thanks for spotting this.

The proper fix is modifying the following method in `ORKGraphChartView.m`:

```
- (void)tintColorDidChange {
    _xAxisView.tintColor = self.tintColor;
    [self updatePlotColors];
}
```

I have included this fix as a part of the ongoing [Pull Request #160](https://github.com/ResearchKit/ResearchKit/pull/620).

**Comment by rsanchezsaez** [comment]
This fix has been merged and the issue can be closed.
